src/main/kotlin/no/nav/syfo/client/person/adressebeskyttelse/AdressebeskyttelseClient.kt

```diff
@@ -14,34 +16,41 @@ import no.nav.syfo.util.*
 import org.slf4j.LoggerFactory
 
 class AdressebeskyttelseClient(
-    syfopersonBaseUrl: String
+    private val cache: RedisStore,
+    private val httpClient: HttpClient = httpClientDefault(),
+    syfopersonBaseUrl: String,
 ) {
-    private val personAdressebeskyttelseUrl: String
-
-    init {
-        this.personAdressebeskyttelseUrl = "$syfopersonBaseUrl$PERSON_ADRESSEBESKYTTELSE_PATH"
-    }
-
-    private val httpClient = httpClientDefault()
+    private val personAdressebeskyttelseUrl: String = "$syfopersonBaseUrl$PERSON_ADRESSEBESKYTTELSE_PATH"
 
     suspend fun hasAdressebeskyttelse(
         personIdentNumber: PersonIdentNumber,
         token: String,
         callId: String
     ): Boolean {
-        return try {
-            val response: HttpResponse = httpClient.get(personAdressebeskyttelseUrl) {
-                header(HttpHeaders.Authorization, bearerHeader(token))
-                header(NAV_CALL_ID_HEADER, callId)
-                header(NAV_PERSONIDENT_HEADER, personIdentNumber.value)
-                accept(ContentType.Application.Json)
-            }
-            COUNT_CALL_PERSON_ADRESSEBESKYTTELSE_SUCCESS.inc()
-            response.receive<AdressebeskyttelseResponse>().beskyttet
-        } catch (e: ClientRequestException) {
-            handleUnexpectedResponseException(e.response, callId)
-        } catch (e: ServerResponseException) {
-            handleUnexpectedResponseException(e.response, callId)
+        val cacheKey = "$CACHE_ADRESSEBESKYTTELSE_KEY_PREFIX${personIdentNumber.value}"
+        return when (val cachedAdresseBeskyttelse = cache.get(cacheKey)) {
```

Comment:
Kanskje flytte assign ut av when, leser litt rart i mitt hode :) (val cachedAdresseBeskyttelse = cache.get(cacheKey))

Reply:
IntelliJ hadde faktisk en warning på at den kunne flyttes inn, men for så vidt enig i at det blir ryddigere utenfor :)